### DISCUSSION TITLE
Control of middleware execution order

### DISCUSSION THREAD

**Discussion by ghost** [nitrojs/nitro#735] [Q&A]
Dears, 

I wonder if there is a possibility to define middleware execution order.  Like this
1. /middleware/auth.ts (adding user to context)
2. /middleware/protect.ts (protect route by throwing 403 if context.user.role is not eligible for the route)
...
7. 
Regards TJ

**Comment by misaon** [comment]
+1

**Comment by ghost** [comment]
So far, come to very simple workaround 

```ts
// ~/server/middleware/index.ts
import authMiddleware from '../handlers/auth';
import logMiddleware from '../handlers/log';

export default defineEventHandler(event => {
  logMiddleware(event);
  authMiddleware(event);
});

```

```ts
// ~/server/handlers/log.ts
const logMiddleware = defineEventHandler(event => {
  if (!event.path?.includes('/api/v1')) {
    return;
  }

  console.log(event.node.req.method, event.path);
});
export default logMiddleware;

```

```ts
// ~/server/handlers/auth.ts
const authMiddleware = defineEventHandler(event => {
  event.context.user='foo'
});
export default authMiddleware;

```

~/server/middleware/index.ts can be ``` (async event=>{})```  as well.
Changing the order of the calls in index.ts solves the question. 

I know it is simple - usually, I  like simple solutions. Please tell me maybe I underseen something.
